subsys/bluetooth/mesh/settings.c

```diff
@@ -111,12 +117,12 @@ void bt_mesh_settings_store_schedule(enum bt_mesh_settings_flag flag)
 
 	if (atomic_get(pending_flags) & NO_WAIT_PENDING_BITS) {
 		timeout_ms = 0;
-	} else if (CONFIG_BT_MESH_RPL_STORE_TIMEOUT >= 0 &&
+	} else if (IS_ENABLED(CONFIG_BT_MESH_RPL_STORAGE_MODE_SETTINGS) && RPL_STORE_TIMEOUT >= 0 &&
 		   atomic_test_bit(pending_flags, BT_MESH_SETTINGS_RPL_PENDING) &&
 		   !(atomic_get(pending_flags) & GENERIC_PENDING_BITS)) {
-		timeout_ms = CONFIG_BT_MESH_RPL_STORE_TIMEOUT * MSEC_PER_SEC;
+		timeout_ms = RPL_STORE_TIMEOUT * MSEC_PER_SEC;
 	} else {
-		timeout_ms = CONFIG_BT_MESH_STORE_TIMEOUT * MSEC_PER_SEC;
+		timeout_ms = RPL_STORE_TIMEOUT * MSEC_PER_SEC;
```

Comment:
Doesn't this potentially expand to `-1 * MSEC_PER_SEC`?

Reply:
You are right, it is a bug. Should be `BT_MESH_STORE_TIMEOUT`